google/cloud/security/common/data_access/project_dao.py

```diff
@@ -33,15 +33,41 @@
 LOGGER = log_util.get_logger(__name__)
 
 
-class ProjectDao(_db_connector.DbConnector):
+class ProjectDao(dao.Dao):
     """Data access object (DAO)."""
 
     def __init__(self):
         super(ProjectDao, self).__init__()
 
+    def get_project_numbers(self, resource_name, timestamp):
+        """Select the project numbers from a projects snapshot table.
+
+        Args:
+            resource_name: String of the resource name.
+            timestamp: String of timestamp, formatted as YYYYMMDDTHHMMSSZ.
+
+        Returns:
+             list of project numbers
+
+        Raises:
+            MySQLError: An error with MySQL has occurred.
+        """
+        try:
+            project_numbers_sql = select_data.PROJECT_NUMBERS.format(timestamp)
+            rows = self.execute_sql_with_fetch(
+                resource_name, project_numbers_sql, ())
+            return [row['project_number'] for row in rows]
+        except (DataError, IntegrityError, InternalError, NotSupportedError,
```

Comment:
You can actually do without the try/except as self.execute_sql_with_fetch() will already do that, and raise MySQLError, which you can handle here or let it bubble up and handled higher up.